osu.Game/Configuration/DiscordRichPresenceMode.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.ComponentModel;
+
+namespace osu.Game.Configuration
+{
+    public enum DiscordRichPresenceMode
+    {
+        Disabled,
+
+        [Description("Hide identifiable information")]
+        Limited,
+
+        Enabled
```

Comment:
Maybe `Off` and `Full` would be better than `Disabled`/`Enabled`? To fit with the ternary middle value better.

Reply:
On second thought, I think this would look a bit weird to users. Should descriptions be added? Maybe something like this:
- Disabled
- Show only non-identifiable information
- Show all information

Reply:
`Off`/`Only non-identifiable information`/`Full` could maybe work? Although "non-identifiable information" is also sort of opaque...

Sometimes services will refer to the "non-identifiable" mode as to "streamer mode" but again that's a little weird to use in this context.

Generally not sure of the usefulness of the "limited" mode. Might as well turn off entirely at that point?

Reply:
> Generally not sure of the usefulness of the "limited" mode. Might as well turn off entirely at that point?

I'm mostly implementing up to the linked issue. We could indeed implement only a simple boolean toggle instead of multiple modes, though I personally think that players should be given the option to show off what they're playing without being potentially followed around by random people.

Reply:
I think there's definitely an audience for the limited mode.

Reply:
Hm, fair enough. But yeah, should keep the names consistent whichever way it goes. It should either be `Hide`/`Show only non-identifiable information`/`Show all information`, or the version I proposed above, or something. Having `Disabled` in there alongside two `Show`s seems super off.

Reply:
For sure. Has already been updated to something I can call acceptable, check latest commit.

Reply:
Ah yeah I actually missed the fact that that got applied entirely.